src/hist/core.py

```diff
@@ -11,19 +11,26 @@
 class BaseHist(Histogram):
     def __init__(self, *args, **kwargs):
         """
-            Initialize Hist object. Axis params must contain the names.
+            Initialize BaseHist object. Axis params must contain the names.
         """
 
         super().__init__(*args, **kwargs)
         self.names: dict = dict()
         for ax in self.axes:
             if ax.name in self.names:
                 raise Exception(
-                    "Hist instance cannot contain axes with duplicated names."
+                    "BaseHist instance cannot contain axes with duplicated names."
                 )
             else:
                 self.names[ax.name] = True
 
+    def project(self, *args: Tuple[int]):
```

Comment:
How to deal with the pre-commit failure.

Reply:
`Tuple[Union[str, int]]` is the correct signature here. We may have to disable this one check on one line for NamedHist, as it will have more restrictive type signatures than Hist, which it's based on. There's a way to do that which I can look up.

Reply:
Okay. But it's incompatible with `NamedHist.fill()`, where users are only allowed to fill by names.